test/spec/modules/beopBidAdapter_spec.js

```diff
@@ -0,0 +1,135 @@
+      let bid = Object.assign({}, validBid);
+      delete bid.params;
+      bid.params = {
+        'someId': '12345'
+      };
+      expect(spec.isBidRequestValid(bid)).to.equal(false);
+    });
+
+    it('should return false if there is no banner media type', function () {
+      let bid = Object.assign({}, validBid);
+      delete bid.mediaTypes;
+      bid.mediaTypes = {
+        'native': {
+          'sizes': [[1, 1]]
+        }
+      };
+      expect(spec.isBidRequestValid(bid)).to.equal(false);
+    });
+  });
+
+  describe('buildRequests', function () {
+    let bidRequests = [];
+    bidRequests.push(validBid);
+
+    it('should build the request', function () {
+      const request = spec.buildRequests(bidRequests, {});
+      const payload = JSON.parse(request.data);
+      const url = request.url;
+      expect(url).to.equal(ENDPOINT);
+      expect(payload.pid).to.exist;
+      expect(payload.pid).to.equal('5a8af500c9e77c00017e4cad');
+      expect(payload.slts[0].name).to.exist;
+      expect(payload.slts[0].name).to.equal('bellow-article');
+    });
+
+    it('should call the endpoint with GDPR consent and pageURL info if found', function () {
+      let consentString = 'BOJ8RZsOJ8RZsABAB8AAAAAZ+A==';
+      const request = spec.buildRequests(bidRequests, {});
+    });
+  });
+
+  describe('timeout and bid won pixel trigger', function () {
+    let triggerPixelStub;
+
+    beforeEach(function () {
+      triggerPixelStub = sinon.stub(utils, 'triggerPixel');
+    });
+
+    afterEach(function () {
+      utils.triggerPixel.restore();
+    });
+
+    it('should call triggerPixel utils function when timed out is filled', function () {
+      spec.onTimeout({});
+      spec.onTimeout();
+      expect(triggerPixelStub.getCall(0)).to.be.null;
+      spec.onTimeout({params: {accountId: '5a8af500c9e77c00017e4cad'}, timeout: 2000});
+      expect(triggerPixelStub.getCall(0)).to.not.be.null;
+      expect(triggerPixelStub.getCall(0).args[0]).to.exist.and.to.include('https://t.beop.io');
+      expect(triggerPixelStub.getCall(0).args[0]).to.include('se_ca=bid');
+      expect(triggerPixelStub.getCall(0).args[0]).to.include('se_ac=timeout');
+      expect(triggerPixelStub.getCall(0).args[0]).to.include('pid=5a8af500c9e77c00017e4cad');
+    });
+
+    it('should call triggerPixel utils function on bid won', function () {
+      spec.onBidWon({});
+      spec.onBidWon();
+      expect(triggerPixelStub.getCall(0)).to.be.null;
+      spec.onBidWon({params: {accountId: '5a8af500c9e77c00017e4cad'}, cpm: 1.2});
+      expect(triggerPixelStub.getCall(0)).to.not.be.null;
+      expect(triggerPixelStub.getCall(0).args[0]).to.exist.and.to.include('https://t.beop.io');
+      expect(triggerPixelStub.getCall(0).args[0]).to.include('se_ca=bid');
+      expect(triggerPixelStub.getCall(0).args[0]).to.include('se_ac=won');
+      expect(triggerPixelStub.getCall(0).args[0]).to.exist.and.to.include('pid=5a8af500c9e77c00017e4cad');
+    });
+  });
+});
```

Comment:
all of your tests are on the request, perhaps you would like a test on the response?

Reply:
Hello, thanks for the comment. For now, our server is responsible for respect the prebid output format for the responses. So the handler on prebid.js code is just acting as a pass-through from our server to responses. If we add some transformation one day, we will surely add tests on the response. 
If this is mandatory for you to have these test, we can eventually mock a server response, that already fits the prebid response format to be sure that our handler is not breaking anything, but once again, we are not doing any transformation. Let me know

Reply:
Hello @patmmccann, I have added tests as you mentioned it about responses. Please feel free to give us a feedback